packages/mdctl-api/index.d.ts

```diff
@@ -0,0 +1,40 @@
+export class Environment {
+
+  constructor(input: string)
+  url: string
+
+}
+
+export class Client {
+
+  environment: Environment
+
+  constructor(params: {
+    environment:
+      | {
+      endpoint: string
+      env: string
+    }
+      | string
+    credentials: {
+      type?: string
+      token?: string
+      apiKey?: string
+    }
+    sessions: boolean
+    requestOptions: {
+      strictSSL: boolean
+    }
+  })
+
+  put<R = unknown, P = unknown>(url: string, params: P): Promise<R>
+  delete<R = unknown>(url: string): Promise<R>
+  post<R = unknown, P = unknown, P2 = unknown>(
+    url: string,
+    params: P,
+    param?: P2
```

Comment:
If there's no specific need to name this parameter `param`, I'd like it to be something different, because if not it's too similar to the other argument, `params`. What about `payload`?